types/block.go

```diff
@@ -99,6 +103,13 @@ func (b Block) ID() BlockID {
 	return BlockID(crypto.HashObject(b.Header()))
 }
 
+// ID returns the ID of a Block, which is calculated by hashing the
+// concatenation of the block's parent's ID, nonce, and the result of the
+// b.MerkleRoot().
+func (h BlockHeader) ID() BlockID {
```

Comment:
The comment here is confusing. While correct, the logic just hashes the header, which doesn't seem to match the comment.